### DISCUSSION TITLE
How to submit a Gatsby plugins security vulnerabilities patch?

### DISCUSSION THREAD

**Discussion by jpventura** [gatsbyjs/gatsby#31264] [Help]
# TL; DR
Modifications submitted on PR #31251

# Details
I was updating a Gatsby Starter package and it was not possible running `npm audit fix` due a circular dependency on `gatsby`. 

Also I could not figure out how to run the local tests only on the problematic packages, but (according [Circle CI report](https://app.circleci.com/pipelines/github/gatsbyjs/gatsby/62674/workflows/0cc29f84-e9c5-4c44-8cab-73705b5a99d2)), only [`e2e_tests_pnp` tests failed](https://app.circleci.com/pipelines/github/gatsbyjs/gatsby/62674/workflows/0cc29f84-e9c5-4c44-8cab-73705b5a99d2/jobs/685768).

My starter direct dependencies are:

  - `gatsby@3.4.1`
  - `gatsby-image@3.4.0`
  - `gatsby-plugin-manifest@3.4.0`
  - `gatsby-plugin-material-ui@3.0.1`
  - `gatsby-plugin-offline@4.4.0`
  - `gatsby-plugin-react-helmet@4.4.0`
  - `gatsby-plugin-sharp@3.4.1`
  - `gatsby-source-filesystem@3.4.0`
  - `gatsby-transformer-sharp@3.4.0`

# NPM Audit Report

The data from this report was generated by `npm audit --json` and the packages linked to [Snyk I/O](https://snyk.io) as possible.

| name | severity | fix available |
|------|----------|---------------|
| `babel-plugin-remove-graphql-queries` | high | [`babel-plugin-remove-graphql-queries@2.14.0`](https://snyk.io/advisor/npm-package/babel-plugin-remove-graphql-queries#versions) |
| `engine.io-client` | high | [`engine.io-client@5.0.1`](https://snyk.io/advisor/npm-package/engine.io-client#versions) |
| [`gatsby`](https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby) | high | [`gatsby@2.22.10`](https://snyk.io/advisor/npm-package/gatsby#versions) |
| [`gatsby-plugin-manifest`](https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby-plugin-manifest) | high | [`gatsby-plugin-manifest@2.6.1`](https://snyk.io/advisor/npm-package/gatsby-plugin-manifest#versions) |
| [`gatsby-plugin-material-ui`](https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby-plugin-material-ui) | high | [`gatsby-plugin-material-ui@2.1.10`](https://snyk.io/advisor/npm-package/gatsby-plugin-material-ui#versions) |
| [`gatsby-plugin-offline`](https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby-plugin-offline) | high | [`gatsby-plugin-offline@3.10.2`](https://snyk.io/advisor/npm-package/gatsby-plugin-offline#versions) |
| [`gatsby-plugin-page-creator`](https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby-plugin-page-creator) | high | [`undefined@undefined`](https://snyk.io/advisor/npm-package/gatsby-plugin-page-creator#versions) |
| [`gatsby-plugin-react-helmet`](https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby-plugin-react-helmet) | high | [`gatsby-plugin-react-helmet@3.10.0`](https://snyk.io/advisor/npm-package/gatsby-plugin-react-helmet#versions) |
| [`gatsby-plugin-sharp`](https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby-plugin-sharp) | high | [`gatsby-plugin-sharp@2.14.4`](https://snyk.io/advisor/npm-package/gatsby-plugin-sharp#versions) |
| [`gatsby-plugin-typescript`](https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby-plugin-typescript) | high | |
| [`gatsby-plugin-utils`](https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby-plugin-utils) | high | [`gatsby@2.22.10`](https://snyk.io/advisor/npm-package/gatsby-plugin-utils#versions) |
| [`gatsby-source-filesystem`](https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby-source-filesystem) | high | [`gatsby-source-filesystem@2.11.1`](https://snyk.io/advisor/npm-package/gatsby-source-filesystem#versions) |
| [`gatsby-transformer-sharp`](https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby-transformer-sharp) | high | [`gatsby-transformer-sharp@2.12.1`](https://snyk.io/advisor/npm-package/gatsby-transformer-sharp#versions) |
| `socket.io-client` | high | [`socket.io-client@4.0.1`](https://snyk.io/advisor/npm-package/socket.io-client#versions) |
| `xmlhttprequest-ssl` | high | [`xmlhttprequest-ssl@1.6.2`](https://snyk.io/advisor/npm-package/xmlhttprequest-ssl#versions) |

**Comment by LekoArts** [comment]
Please see https://github.com/gatsbyjs/gatsby/issues/31266#issuecomment-833319227 for why you need to take such reports with a grain of salt. Additionally, I don't see how e.g. gatsby-plugin-react-helmet has a "high" severity? Synk doesn't show anything -- so not sure what you're trying to say here :)
